File: server-api/server-api-go/web.go
// author admin@elesos.com

//请参考：https://docs.starrtc.com/en/docs/aec-index.html

package main

import (
       "fmt"
       "github.com/gin-gonic/gin"
       "net/http"
       "io/ioutil"
       "strings"
       "encoding/json"
       "crypto/hmac"
       "crypto/sha1"
       "encoding/base64"
)

func main() {
  fmt.Printf("Let's go...")
  router := gin.Default()
    
  
  router.GET("/eventCenter", func(c *gin.Context) {
    data := c.DefaultQuery("data", "")
    if data == "" {
      c.JSON(200, gin.H{
        "status":0,
        "data":"invalid args",
      })
      return
    }    
        
    
    var data_obj map[string]interface{} 
    json.Unmarshal([]byte(data), &data_obj) 
    
	
    if "AEC_GROUP_CREATE" == data_obj["action"] {
     c.JSON(200, gin.H{
          "status":0,
          "data":"you have no rights to create group",
        })
        return
    }
    
	//TODO process other actions
	
	
	
	
    c.JSON(200, gin.H{
         "status":0,
         "data":"unkown action",
       })
       return
  })
 
  router.Run(":9091")
}

File: server-api/server-api-node.js/web.js
// author: admin@elesos.com

//请参考：https://docs.starrtc.com/en/docs/aec-index.html

const Koa 		 = require('koa');
const bodyParser = require('koa-bodyparser');
const router 	 = require('koa-router')();
const request 	 = require('request');

const app 	     = new Koa();
const crypto 	 = require('crypto');

// parse request body
app.use(bodyParser());


//log request url
app.use(async (ctx, next) => {
  console.log(`Process ${ctx.request.method} ${ctx.request.url}...`);
  await next();
});




router.get('/eventCenter', async (ctx, next) => {
  var data = ctx.request.query.data || null;  
  ctx.response.type = 'json';
  if(!data){
    ctx.response.body = echo_0('invalid args');
    return;
  }
   
  json_obj = JSON.parse(data);

  
  if('AEC_GROUP_CREATE' === json_obj.action){
	//TODO
    var groupId = 10010;
    ctx.response.body = echo_1(groupId);
    return;
  }
  // TODO process other actions
  
  
  ctx.response.body = echo_0('unkown action'); 
  return;
});


function echo_0(data){
  var obj = {"status":0,"data":data};
  return JSON.stringify(obj);
}

function echo_1(data){
  var obj = {"status":1,"data":data};
  return JSON.stringify(obj);
}

//test
router.get('/hello/:name', async (ctx, next) => {
  var name = ctx.params.name;
  ctx.response.body = `<h1>Hello, ${name}!</h1>`;
});



app.use(router.routes());

app.listen(9090);
console.log('app started at port 9090...');
